cypress/integration/checkout/checkout.multipiece.items.spec.js

```diff
@@ -0,0 +1,145 @@
+          barcode: Helper.getRandomBarcode(),
+          status:  { name: 'Available' },
+          permanentLoanType: { id: Cypress.env('loanTypes')[0].id },
+          materialType: { id: materialTypeName.id },
+        };
+        if (numberOfPieces) {
+          defaultItem.numberOfPieces = numberOfPieces;
+        }
+        if (hasDiscription) {
+          defaultItem.descriptionOfPieces = defautlDescription;
+        }
+        if (hasMissingPieces) {
+          defaultItem.numberOfMissingPieces = 2;
+          defaultItem.missingPieces = defautlDescription;
+        }
+        return defaultItem;
+      };
+      testItems.push(getTestItem(1, false, false));
+      testItems.push(getTestItem(3, true, false));
+      testItems.push(getTestItem(2, true, true));
+      testItems.push(getTestItem(1, false, true));
+      InventoryInstances.createFolioInstanceViaApi({
+        instance: {
+          instanceTypeId: Cypress.env('instanceTypes')[0].id,
+          title: instanceTitle,
+        },
+        holdings: [{
+          holdingsTypeId: Cypress.env('holdingsTypes')[0].id,
+          permanentLocationId: Cypress.env('locations')[0].id,
+        }],
+        items: testItems,
+      })
+        .then(specialInstanceIds => {
+          testInstanceIds = specialInstanceIds;
+        });
+    });
+
+    cy.createTempUser([
+      permissions.checkoutCirculatingItems.gui,
+    ])
+      .then(userProperties => {
+        user = userProperties;
+        servicePoint = NewServicePoint.getDefaulServicePoint();
+        ServicePoints.createViaApi(servicePoint.body);
+        UsersEditPage.addServicePointsToUser([servicePoint.body.id],
+          user.userId, servicePoint.body.id);
+      })
+      .then(() => {
+        cy.getUsers({ limit: 1, query: `"personal.lastName"="${user.username}" and "active"="true"` })
+          .then((users) => {
+            userBarcode = users[0].barcode;
+          });
+        cy.login(user.username, user.password, { path: TopMenu.checkOutPath, waiter: Checkout.waitLoading });
+      });
+  });
+
+  after(() => {
+    testItems.forEach(item => {
+      CheckInActions.checkInItem(item.barcode);
+    });
+    cy.wrap(testInstanceIds.holdingIds.forEach(holdingsId => {
+      cy.wrap(holdingsId.itemIds.forEach(itemId => {
+        cy.deleteItem(itemId);
+      })).then(() => {
+        cy.deleteHoldingRecord(holdingsId.id);
+      });
+    })).then(() => {
+      cy.deleteInstanceApi(testInstanceIds.instanceId);
+    });
+    SwitchServicePoint.changeServicePointPreference(user.username);
+    cy.deleteServicePoint(servicePoint.body.id);
+    Users.deleteViaApi(user.userId);
+  });
+
+  it('C591 Check out: multipiece items', { tags: [TestTypes.smoke] }, () => {
+    CheckOutActions.checkIsInterfacesOpened();
+    CheckOutActions.checkOutItem(userBarcode, testItems[0].barcode);
+    CheckOutActions.checkPatronInformation(user.username, userBarcode);
+    CheckOutActions.checkConfirmMultipieceCheckOut();
```

Comment:
add isPresented into checkConfirmMultipieceCheckOut  to check in one place availability or absence

Reply:
Done